v2/dtos/reading.go

```diff
@@ -18,12 +18,11 @@ import (
 type BaseReading struct {
 	common.Versionable `json:",inline"`
 	Id                 string   `json:"id"`
-	Pushed             int64    `json:"pushed,omitempty"`
 	Created            int64    `json:"created"`
 	Origin             int64    `json:"origin"`
 	Modified           int64    `json:"modified,omitempty"`
-	DeviceName         string   `json:"deviceName"`
-	Name               string   `json:"name"`
+	DeviceName         string   `json:"deviceName" validate:"required"`
+	Name               string   `json:"name" validate:"required"`
 	Labels             []string `json:"labels,omitempty"`
 	BinaryReading      `json:",inline"`
```

Comment:
`FloatEncoding` is required if `ValueType` = `ValueTypeFloat32` or  `ValueTypeFloat64`

Reply:
Also should `ValueType` be validated for accepted values? That is a long list and not done in V1's validation. Just a question that we should consider.

Reply:
> `FloatEncoding` is required if `ValueType` = `ValueTypeFloat32` or `ValueTypeFloat64`

Added the check in `func (b BaseReading) Validate()`.